File: _merge_intervals.ts
function merge(intervals: number[][]): number[][] {
    let res = []

    intervals.sort((a,b)=>a[0]-b[0])
    
    res.push(intervals[0])

    for(let cur of intervals){
        let curStart = cur[0]
        let curEnd = cur[1]
        let prevEnd = res[res.length-1][1]

        if(curStart<=prevEnd){
            res[res.length-1][1] = Math.max(curEnd,prevEnd)
        }
        else{
            res.push(cur)
        }
    }
    
    return res

};

File: Python/merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:

        res = [] 
        intervals = sorted(intervals, key=lambda x:x[0])
        # print('intervals',intervals)
        res.append(intervals[0])

        for i in range(1,len(intervals)):
            prev = res.pop()
            # print('prev',prev)
            cur = intervals[i]
            if(cur[0]>prev[1]):
                res.append(prev)
                res.append(cur)
            else:
                first = min(prev[0],cur[0])
                second = max(prev[1], cur[1])
                res.append([first, second])
        
        return res
